### ISSUE TITLE
Add default flags in --help output

### ISSUE THREAD

**Issue by tal-rofe** [jmcdo29/nest-commander#780]
### Is there an existing issue that is already proposing this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe it

Currently, when adding such command:

```ts
@Command({
	name: '-',
	description: 'Command handler for base CLI',
	options: { isDefault: true, hidden: true },
})
export class BaseCommand extends CommandRunner {
	public async run(_: string[], options?: ICommandOptions) {
		if (options?.version) {
			render(<Text>v{__VERSION__}</Text>);

			process.exit(0);
		}

		if (options?.envInfo) {
			const [nodeJsVersion, npmJsVersion] = await Promise.all([getNodeJsVersion(), getNpmVersion()]);

			render(
				<Box display="flex" flexDirection="column">
					<Text>NodeJS version: v{nodeJsVersion}</Text>
					<Text>NPM version: v{npmJsVersion}</Text>
					<Text>
						Operation System: {os.platform()} {os.release()}
					</Text>
				</Box>,
			);

			process.exit(0);
		}

		await Promise.resolve();

		process.exit(0);
	}

	@Option({
		flags: '-v, --version [version]',
		description: 'Return version of the Exlint CLI',
		required: false,
	})
	public version(value: unknown) {
		return value;
	}

	@Option({
		flags: '--env-info [envInfo]',
		description: 'Output execution environment information',
		required: false,
	})
	public environmentInfo(value: unknown) {
		return value;
	}
}

```

If I do `MY_CLI -h` it outputs: 
![image](https://user-images.githubusercontent.com/100444463/218046146-b894e15c-bc64-4aeb-935a-1ef1299215f8.png)

That means, without my default flags.

So 2 things would be nice here:
1. Remove the required `name` from `@command` decorator, **in case this command is marked as the default one.**
So I should be able to do:
```ts
@Command({
	description: 'Command handler for base CLI',
	options: { isDefault: true, hidden: true },
})
```
2. add any flags added in a default command to the `--help` output

### Describe the solution you'd like

=---

### What is the motivation / use case for changing the behavior?

---

**Comment by jmcdo29** [comment]
I would rather not remove the `name` due to the fact of how commands are normally created and added and the complexity that that would bring. I might be able to check for the `options.isDefault` and automatically add the name `default` or `.` or something, but the name would still exist. It would probably be possible to attach the rest of the command metadata to the `program` itself, but it would be added complexity in the code which would lead to a maintenance debt, I think.

The rest of this should be relatively feasible though, and I'll try to get to working on it this week

**Comment by tal-rofe** [comment]
> I would rather not remove the `name` due to the fact of how commands are normally created and added and the complexity that that would bring. I might be able to check for the `options.isDefault` and automatically add the name `default` or `.` or something, but the name would still exist. It would probably be possible to attach the rest of the command metadata to the `program` itself, but it would be added complexity in the code which would lead to a maintenance debt, I think.
> 
> The rest of this should be relatively feasible though, and I'll try to get to working on it this week

Well, using TS and configure `name` to be invalid when providing `option.isDefault` to `true` should be easy. I think such enforcement is required for that scenario. You don't want to name a command, when it is dedicated for the default one.

**Comment by jmcdo29** [comment]
The `name` attribute shouldn't be invalid when `isDefault` is `true` though. You should still be able to call `<program> <name> <arguments>` just like you'd call `<program> <arguments>`, they should be the same thing in this case

**Comment by tal-rofe** [comment]
> The `name` attribute shouldn't be invalid when `isDefault` is `true` though. You should still be able to call `<program> <name> <arguments>` just like you'd call `<program> <arguments>`, they should be the same thing in this case

So you want to allow a user to easily create command: `cli command --version` and `cli --version` with same functionality. Ok, I get it. But at-least allow `name` to be `undefined`?

**Comment by jmcdo29** [comment]
I can probably work out the types to allow name to be `undefined` on the `@Command(0` and populate it with something like `.`/`-`/`$` under the hood. 

The other thing I _might_ try to implement instead is instead of creating a new `Command` instance to attach to the commander `program`, just take the command that _would_ be created and make that the `program` itself. I'll have to think around how much code that would introduce but I think it can mostly be re-used

**Comment by tal-rofe** [comment]
@jmcdo29 
In my opinion, you could keep the `@Command` as is, and create a new decorator `@DefaultCommand` to apply only flags on the program itself.

**Comment by jmcdo29** [comment]
I was thinking of calling it `@RootCommand()`, but `@DefaultCommand()` works too. It'll need to be treated differently either way

**Comment by jmcdo29** [comment]
[Published in 3.6.0](https://nest-commander.jaymcdoniel.dev/en/features/commander/#using-a-command-as-the-root-command)
